### PR TITLE
Added SSRF prevention agent and tests

### PR DESCRIPTION
**Pull request by duvld**
# The problem
Certain vulnerabilities in the way Enketo was parsing media GET and HEAD requests allowed for SSRF attacks (see more here): https://github.com/kobotoolbox/tasks/issues/339

# The changes
Added a `request-filtering-agent` (https://www.npmjs.com/package/request-filtering-agent) to `media-controller.js` to filter out GET and HEAD requests from private IP addresses. This package came with some more filtering options that I incorporated into the filter. These options can be selected in a `config.json`

I _attempted_ to create an exhaustive test suit in `test/server/media-controller.spec.js` to test the different combinations of these filters and if they behave in the expected way or not

Another minor change (that can be ignored) is for some reason when using VIM to make changes, the logs get filled with not being able to parse a `.swp` file. To fix this I made `express.js` ignore `.swp` files 

# Things to do
Check if there's more things to test/keep track of

### CODE REVIEW

**Comment by MartijnR** [comment]
Thank you! Please ping me if I haven't responded on Tuesday 4/14.

**Comment by MartijnR** [comment]
In the mean time, could please fix the tests (see Travis)? :)

File: app/controllers/media-controller.js [hunk 1]
```diff
@@ -9,6 +9,10 @@ const request = require( 'request' );
 const express = require( 'express' );
 const router = express.Router();
 const debug = require( 'debug' )( 'media-controller' );
+const { RequestFilteringHttpAgent } = require( 'request-filtering-agent' );
+
+const config = require('../../config/config');
```
Comment (MartijnR) [inline app/controllers/media-controller.js:14]: Use `require('../models/config-model').server`. Configs can also be created with environment variables, and there is some cleaning/manipulation done in the config-model.

File: test/server/media-controller.spec.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
+const chai = require( 'chai' );
+const expect = chai.expect;
+const request = require( 'request' );
+const config = require( '../../config/config' );
```
Comment (MartijnR) [inline test/server/media-controller.spec.js:4]: (../../app/models/config-model).server

File: config/express.js [hunk 3]
```diff
@@ -102,7 +102,7 @@ app.use( function( req, res, next ) {
 
 // load controllers (including their routers)
 fs.readdirSync( controllersPath ).forEach( function( file ) {
-    if ( file.indexOf( '-controller.js' ) >= 0 ) {
+    if ( file.indexOf( '-controller.js' ) >= 0 && !(file.indexOf( '-controller.js.swp' ) >= 0 )) {
```
Comment (MartijnR) [inline config/express.js:105]: What is `*-controller.js.swp`?

File: app/controllers/media-controller.js [hunk 4]
```diff
@@ -9,6 +9,10 @@ const request = require( 'request' );
...
+const requestFilteringOptions = config.IPfiltering;
```
Comment (MartijnR) [inline app/controllers/media-controller.js:15]: Is there a file missing perhaps? I don't see where `config.IPFiltering` is populated.

File: config/express.js (hunk 3 above)
Comment (MartijnR) [reply config/express.js:105]: Sorry, I didn't read what you wrote about that. That should be taken out. ;)

File: app/controllers/media-controller.js (hunk 4 above)
Comment (duvld) [reply app/controllers/media-controller.js]: It's being populated in the `config.json`, it's where the options for `request-filtering-agent` are -- I've added it to the `default-config.json`.

File: tutorials/00-getting-started.md [hunk 5]
```diff
@@ -48,6 +48,8 @@ The default production configuration includes 2 redis instances. You can **great
 
 For development usages, it is helpful to set "linked form and data server" -> "server url" to `""`, so you can use any OpenRosa server with your local Enketo Express.
 
+To prevent SSRF attacks, a `IPFiltering` object is added to the `default-config.json` configuration that prevents private IP addresses from making GET and HEAD requests.
+
```
Comment (MartijnR) [inline tutorials/00-getting-started.md:52]: Maybe just make it really easy and tell devs exactly how you recommend they help running Enketo Express locally to develop on, i.e. what setting to tweak in config.json during development.

**Pull request closed without merging**